pkg/controller/migration/convert/network.go

```diff
@@ -138,6 +138,11 @@ func handleCalicoCNI(c *components, install *operatorv1.Installation) error {
 			if err := c.node.assertEnv(ctx, c.client, containerCalicoNode, "FELIX_IPV6SUPPORT", "false"); err != nil {
 				return err
 			}
+			// If IP6=none then if IP6_AUTODETECTION_METHOD is set it must be none. This is not a valid value
+			// for the env var but Kops sets it.
+			if err := c.node.assertEnv(ctx, c.client, containerCalicoNode, "IP6_AUTODETECTION_METHOD", "none"); err != nil {
```

Comment:
Maybe it could be something like:
```
	env, err := getEnv(ctx, c.client, c.node.Spec.Template.Spec, containerCalicoNode, ComponentCalicoNode, "IP6_AUTODETECTION_METHOD")
	if err != nil {
		return err
	} else if env != nil && *env == "none" {
		c.node.ignoreEnv(containerCalicoNode, "IP6_AUTODETECTION_METHOD")
	}
	if err := c.node.assertEnv(ctx, c.client, containerCalicoNode, "IP6_AUTODETECTION_METHOD", "none"); err != nil {
		return err
	}
```

That way the intent of the comment matches better with the code. We want to give Kops users a pass with their (mis)configuration without asserting that the value is correct.

Reply:
I believe what you have written is what assertEnv does. I don't think I'd want to use a different pattern here since we use assertEnv in many places for exactly this purpose.

Reply:
Do you think the check here is different functionally than the other places it is used?

Reply:
My snippet missed checking for empty values, but I realize now that does not really apply here. If the env does not exist then nil is returned from getEnv. I approved it.